azurerm/resource_arm_container_registry_test.go

```diff
@@ -144,6 +144,32 @@ func TestAccAzureRMContainerRegistry_basicPremium(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMContainerRegistry_basicBasicUpgradePremium(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMContainerRegistry_basicManaged(ri, testLocation(), "Basic")
+	configUpdated := testAccAzureRMContainerRegistry_basicManaged(ri, testLocation(), "Premium")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMContainerRegistryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMContainerRegistryExistsValidateSkuValue("azurerm_container_registry.test", "Basic"),
```

Comment:
we want to be checking the value in the state rather than in the Remote API - as such can we remove this function and check this in the state instead via:

```
resource.TestCheckAttribbute(resourceName, "sku", "Basic"),
```

(same for below):
```
resource.TestCheckAttribbute(resourceName, "sku", "Premium"),
```

(we only tend to check the Remote API when the API is known to be eventually consistent, hence it's overkill checking the API here since the values will be pulled into the state either way)